usdQt/hierarchyModel.py

```diff
@@ -0,0 +1,457 @@
+            self.model.layoutChanged.emit()
+
+    def __init__(self, stage=None, predicate=Usd.TraverseInstanceProxies(
+                    Usd.PrimIsDefined | ~Usd.PrimIsDefined),
+                 parent=None):
+        """Instantiate an QAbstractItemModel adapter for a UsdStage.
+
+        It's safe for the 'stage' to be None if the model needs to be instatiated
+        without knowing the stage its interacting with.
+
+        'predicate' specifies the prims that may be accessed via the model on
+        the stage. A good policy is to be as accepting of prims as possible
+        and rely on a QSortFilterProxyModel to interactively reduce the view.  
+        Changing the predicate is a potentially expensive operation requiring 
+        rebuilding internal indices and tables, making not ideal for interactive
+        filtering.
+        """
+        super(HierarchyBaseModel, self).__init__(parent)
+
+        self.__predicate = predicate
+        self.__stage = None
+        self.__index = None
+        self.__listener = None
+        self.ResetStage(stage)
+
+    def __IsStageValid(self):
+        return self.__stage and self.__stage.GetPseudoRoot()
+
+    def ResetStage(self, stage):
+        """Resets the model for use with a new stage.  
+
+        If the stage isn't valid, this effectively becomes an empty model.
+        """
+        if stage == self.__stage:
+            return
+        self.beginResetModel()
+        self.__stage = stage
+        if not self.__IsStageValid():
+            self.__index = None
+            self.__listener = None
+        else:
+            self.__index = _HierarchyCache(
+                stage.GetPrimAtPath('/'), self.__predicate)
+            self.__listener = Tf.Notice.Register(
+                Usd.Notice.ObjectsChanged, self.__OnObjectsChanged, self.__stage)
+        self.endResetModel()
+
+    def GetPredicate(self):
+        """Return the predicate used to identify prims that should be exposed to the model
+        """
+        return self.__index.GetPredicate()
+
+    def GetRoot(self):
+        """Retrieve the root of the current hierarchy model"""
+        rootProxy = self.__index.GetRoot()
+        return rootProxy.GetPrim() if rootProxy and not rootProxy.expired else None
+
+    def __OnObjectsChanged(self, notice, sender):
+        resyncedPaths = notice.GetResyncedPaths()
+        resyncedPaths = [path for path in resyncedPaths if path.IsPrimPath()]
+
+        if len(resyncedPaths) > 0:
+            with self.__LayoutChangedContext(self):
+                persistentIndices = self.persistentIndexList()
+                indexToPath = {}
+                for index in persistentIndices:
+                    indexProxy = index.internalPointer()
+                    indexPrim = indexProxy.GetPrim()
+                    indexPath = indexPrim.GetPath()
+
+                    for resyncedPath in resyncedPaths:
+                        commonPath = resyncedPath.GetCommonPrefix(path)
```

Comment:
Thank you for catching that.  It wouldn't surprise me if in general we didn't tickle this, but I'll think of a way to add this to the regression tests.